### PR TITLE
Add ErrorHandlerFunc to simplify creating ErrorHandlers

### PR DESCRIPTION
**Pull request by kvrhdn**
This PR adds a convenience adapter to simplify setting the error handler.

Code before:

```go
func main() {
	otel.SetErrorHandler(myErrorHandler{})

	// ...
}

type myErrorHandler struct {}

func (h myErrorHandler) Handle(err error) {
	// your custom error handling
	level.Error(log.Logger).Log("msg", "OpenTelemetry.ErrorHandler", "err", err)
}
```

After:

```go
func main() {
	otel.SetErrorHandler(otel.ErrorHandlerFunc(func(err error) {
		// your custom error handling
		level.Error(log.Logger).Log("msg", "OpenTelemetry.ErrorHandler", "err", err)
	}))

	// ...
}
```

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -10,6 +10,8 @@ This project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.htm
 
 ### Added
 
+- Added `ErrorHandlerFunc` to use a function as `ErrorHandler`. (#2149)
```
Comment (MrAlias) [inline CHANGELOG.md:13]:
```suggestion
- Added `ErrorHandlerFunc` to use a function as an `"go.opentelemetry.io/otel".ErrorHandler`. (#2149)
```

File: error_handler.go [hunk 2]
```diff
@@ -25,3 +25,12 @@ type ErrorHandler interface {
 	// DO NOT CHANGE: any modification will not be backwards compatible and
 	// must never be done outside of a new major release.
 }
+
+// ErrorHandlerFunc is a convenience adapter to allow the use of a function
+// as ErrorHandler.
+type ErrorHandlerFunc func(error)
+
+// Handle implements ErrorHandler.
```
Comment (MrAlias) [inline error_handler.go:33]:
```suggestion
// Handle handles the irremediable error by calling the ErrorHandlerFunc itself.
```

https://dave.cheney.net/practical-go/presentations/qcon-china.html#_always_document_public_symbols)

**Review by MrAlias — APPROVED** [review]
Outside of minor doc changes, I think these changes are a good addition.

**Comment by kvrhdn** [comment]
Thanks for the suggestion @MrAlias!

File: error_handler.go [hunk 3]
```diff
@@ -25,3 +25,12 @@ type ErrorHandler interface {
 	// DO NOT CHANGE: any modification will not be backwards compatible and
 	// must never be done outside of a new major release.
 }
+
+// ErrorHandlerFunc is a convenience adapter to allow the use of a function
+// as an ErrorHandler.
+type ErrorHandlerFunc func(error)
+
+// Handle handles the irremediable error by calling the ErrorHandlerFunc itself.
+func (f ErrorHandlerFunc) Handle(err error) {
```
Comment (pellared) [inline error_handler.go:34]: I would add an "interface assertion"

```go
var _ ErrorHandler = ErrorHandlerFunc(nil)
```

Comment (kvrhdn) [reply error_handler.go:34]: Good call :+1:

**Review by pellared — APPROVED** [review]

**Pull request merged**